# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.cpp
# Commit: cd503f8f9a72
# Full Hash: cd503f8f9a72b8c701a587794f71405b36bcc809
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-20 14:55:47
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 7: Remove Frame::tls. r=lth
#   
#   We remove Frame::tls and fix prologue/epilogue. Runtime, Ion and Baseline are ready for this move already.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D83064
# ==============================================================================

diff -r 270961083337 -r cd503f8f9a72 js/src/jit/MacroAssembler.cpp
--- a/js/src/jit/MacroAssembler.cpp	Tue Oct 20 10:24:47 2020 +0000
+++ b/js/src/jit/MacroAssembler.cpp	Mon Oct 12 08:08:41 2020 +0000
@@ -3148,11 +3148,11 @@
   uint32_t stackAdjust;
   callWithABIPre(&stackAdjust, /* callFromWasm = */ true);
 
-  // The TLS register is used in builtin thunks and must be set, by ABI.
+  // The TLS register is used in builtin thunks and must be set.
   if (tlsOffset) {
     loadPtr(Address(getStackPointer(), *tlsOffset + stackAdjust), WasmTlsReg);
   } else {
-    MOZ_CRASH("This can be only for an unsupported architecture.");
+    MOZ_CRASH("tlsOffset is Nothing only for unsupported abi calls.");
   }
   CodeOffset raOffset = call(
       wasm::CallSiteDesc(bytecode.offset(), wasm::CallSite::Symbolic), imm);
@@ -4072,11 +4072,6 @@
   memoryBarrier(sync.barrierAfter);
 }
 
-void MacroAssembler::loadWasmTlsRegFromFrame(Register dest) {
-  loadPtr(Address(getStackPointer(), framePushed() + wasm::Frame::tlsOffset()),
-          dest);
-}
-
 void MacroAssembler::BranchGCPtr::emit(MacroAssembler& masm) {
   MOZ_ASSERT(isInitialized());
   masm.branchPtr(cond(), reg(), ptr_, jump());